category/basic.lean

```diff
@@ -20,12 +20,26 @@ run_cmd mk_simp_attr `functor_norm
   g <$> (m <$> x) = (g ∘ m) <$> x :=
 (comp_map _ _ _).symm
 
-@[simp] theorem id_map' (x : f α) : (λa, a) <$> x = x := id_map _
+-- @[simp] theorem id_map' (x : f α) : (λa, a) <$> x = x := id_map _
 
 end functor
 
 section applicative
-variables {f : Type u → Type v} [applicative f] [is_lawful_applicative f]
+variables {f : Type u → Type v} [applicative f]
+
+def mmap₂
+  {α₁ α₂ φ : Type u}
+  (g : α₁ → α₂ → f φ) :
+  Π (ma₁ : list α₁) (ma₂: list α₂), f (list φ)
+| (x :: xs) (y :: ys) := (::) <$> g x y <*> mmap₂ xs ys
+| _ _ := pure []
+
+def mmap₂'  (g : α → β → f γ) : list α → list β → f punit
```

Comment:
is this currently used anywhere?

Reply:
Yes in `data.traversable.derive` which I'm merging in now